### DISCUSSION TITLE
Nested array validation

### DISCUSSION THREAD

**Discussion by r3sist** [vimeo/psalm#9826] [Q&A] [answered]
Dear Community,
Do you have any suggestion why `non-empty-array<int, array<string, scalar>>` return type is invalid?

https://psalm.dev/r/8cfc779e82 

Thank you!

**Comment by orklah** [comment, marked as answer]
Psalm is not yet capable of refining the type from inside a loop like that

**Reply by r3sist** [reply]
Thank you!
